source/mcpele/uniform_cubic_sampling.h

```diff
@@ -1,23 +1,35 @@
 #ifndef _MCPELE_UNIFORM_CUBIC_SAMPLING_H__
 #define _MCPELE_UNIFORM_CUBIC_SAMPLING_H__
 
+#include "pele/array.h"
+
 namespace mcpele {
     
 class UniformCubicSampling : public TakeStep {
 protected:
     std::mt19937_64 m_gen;
-    std::uniform_real_distribution<double> m_dist;
+    std::uniform_real_distribution<double> m_dist05;
+    pele::Array<double> m_boxvec;
+    bool m_cubic;
 public:
     virtual ~UniformCubicSampling() {}
-    UniformCubicSampling(const size_t seed=42, const double delta=1)
+    UniformCubicSampling(const size_t seed=42, const pele::Array<double> boxvec={2})
```

Comment:
The default behavior was to assign each coordinate uniformly within [-1,1]. I wanted to keep that default while changing to the new input consisting of only a boxvector. With a 1d boxvec of length 2, this would be the default. On the other hand, I can not imagine that a default is useful here, so we might as well remove it.